16/umbraco-cms/reference/webhooks/README.md

```diff
@@ -165,50 +155,53 @@ builder.WebhookEvents().Clear().AddCms(false);
+using Umbraco.Cms.Core.Notifications;
+using Umbraco.Cms.Core.PublishedCache;
+using Umbraco.Cms.Core.Services;
+using Umbraco.Cms.Core.Sync;
+using Umbraco.Cms.Core.Webhooks;
+
+namespace CommunityDocs.Controllers;
+
 [WebhookEvent("Content Published", Constants.WebhookEvents.Types.Content)]
-public class MyCustomContentPublishedWebhookEvent : WebhookEventContentBase<ContentPublishedNotification, IContent>
+public class MyCustomContentPublishedWebhookEvent(
+    IWebhookFiringService webhookFiringService,
+    IWebhookService webhookService,
+    IOptionsMonitor<WebhookSettings> webhookSettings,
+    IServerRoleAccessor serverRoleAccessor,
+    IApiContentBuilder apiContentBuilder,
+    IPublishedContentCache publishedContentCache): 
+    WebhookEventContentBase<ContentPublishedNotification, IContent>(webhookFiringService, webhookService, webhookSettings, serverRoleAccessor)
 {
-    private readonly IPublishedSnapshotAccessor _publishedSnapshotAccessor;
-    private readonly IApiContentBuilder _apiContentBuilder;
-
-    public MyCustomContentPublishedWebhookEvent(
-        IWebhookFiringService webhookFiringService,
-        IWebhookService webhookService,
-        IOptionsMonitor<WebhookSettings> webhookSettings,
-        IServerRoleAccessor serverRoleAccessor,
-        IPublishedSnapshotAccessor publishedSnapshotAccessor,
-        IApiContentBuilder apiContentBuilder)
-        : base(webhookFiringService, webhookService, webhookSettings, serverRoleAccessor)
-    {
-        _publishedSnapshotAccessor = publishedSnapshotAccessor;
-        _apiContentBuilder = apiContentBuilder;
-    }
-
     public override string Alias => "Umbraco.ContentPublish";
     protected override IEnumerable<IContent> GetEntitiesFromNotification(ContentPublishedNotification notification) => notification.PublishedEntities;
 
     protected override object? ConvertEntityToRequestPayload(IContent entity)
     {
-        if (_publishedSnapshotAccessor.TryGetPublishedSnapshot(out IPublishedSnapshot? publishedSnapshot) is false || publishedSnapshot!.Content is null)
-        {
-            return null;
-        }
-
-        IPublishedContent? publishedContent = publishedSnapshot.Content.GetById(entity.Key);
+        var cachedPublishedContent = publishedContentCache.GetById(entity.Key);
 
         return new
         {
             CustomData = "Your data",
-            PublishedContent = publishedContent is null ? null : _apiContentBuilder.Build(publishedContent)
+            PublishedContent = cachedPublishedContent is null ? null : apiContentBuilder.Build(cachedPublishedContent)
         };
     }
 }
 ```
 
-To replace the default Umbraco webhook with your custom implementation:
+{% hint style="info" %}
+Umbraco developers will need to inject `IPublishedContentCache`, `IPublishedMediaCache`, `IPublishedMemberCache` and `IPublishedContentTypeCache` dependencies individually, instead of injecting the `IPublishedSnapshotAccessor` as would have been done previously.
```

Comment:
```suggestion
Umbraco developers need to inject `IPublishedContentCache`, `IPublishedMediaCache`, `IPublishedMemberCache`, and `IPublishedContentTypeCache` individually, instead of injecting the `IPublishedSnapshotAccessor` as in previous versions.
```